broker/src/main/java/io/zeebe/broker/Broker.java

```diff
@@ -407,10 +407,14 @@ private AutoCloseable partitionsStep(
                     buildExporterRepository(brokerCfg));
             final PartitionTransitionImpl transitionBehavior =
                 new PartitionTransitionImpl(context, LEADER_STEPS, FOLLOWER_STEPS);
-            final ZeebePartition zeebePartition = new ZeebePartition(context, transitionBehavior);
+            final ZeebePartition zeebePartition =
+                new ZeebePartition(
+                    context,
+                    transitionBehavior,
+                    List.of(
+                        new PartitionHealthBroadcaster(
+                            partitionId, topologyManager::onHealthChanged)));
             scheduleActor(zeebePartition);
-            zeebePartition.addFailureListener(
-                new PartitionHealthBroadcaster(partitionId, topologyManager::onHealthChanged));
```

Comment:
@deepthidevaki how do you feel about moving this failure listener to the constructor? I think it makes sense, but perhaps I'm overlooking an actor scheduler issue.

Reply:
It should be safe to move it to the constructor. But what is the need for it?

Reply:
I understood that it solved the second problem @aivinog1 mentioned [here](https://github.com/zeebe-io/zeebe/pull/5790#issuecomment-723980771). I assumed there was a race condition between the partition becoming healthy/unhealthy and when the partition health broadcaster is added as failure listener.

Reply:
@korthout Yes, I tried but without success. Seems that the only way :(

Reply:
Ah yes, there is a timing issue. I dove into it a bit. 

We register the broadcaster as a failure listener using the actor scheduler, but at the moment we call this, we have (in the line before) already activated the actor scheduler which is busy starting the partition. During this start, the health has already been determined (multiple times even) and each failure listener is notified when it changed. Directly after the start, the scheduler does the next task (adding the broadcaster) and after that the health doesn't change any more.

So, I see 2 solutions: 
1. when a new listener is added to the partition, check the health and update the listener accordingly.
2. pass the broadcaster to the partition in the constructor

After discussing with @deepthidevaki solution 1 has our preference, since this pattern is already applied in other places as well.

Reply:
@korthout Many thanks for the analysis :) I'll apply the first solution soon

Reply:
@korthout Can you take a look, please? :) I've also add small unit tests that verify calling `io.zeebe.util.health.FailureListener#onFailure` or `io.zeebe.util.health.FailureListener#onRecovered` depends the partition health.